sdcm/utils/sstable_load_utils.py

```diff
@@ -0,0 +1,162 @@
+import random
+import re
+from collections import namedtuple
+from typing import Any, List, Iterable
+
+from data_dir.test_data_library.load_inventory import TestDataInventory
+from sdcm.keystore import KeyStore
+from sdcm.utils.common import remote_get_file, LOGGER
+from sdcm.utils.decorators import timeout as timeout_decor
+
+
+class SstableLoadUtils:
+    LOAD_AND_STREAM_RUN_EXPR = r'storage_service - load_and_stream:'
+    LOAD_AND_STREAM_DONE_EXPR = r'storage_service - Done loading new SSTables for keyspace={}, table={}, ' \
+                                r'load_and_stream=true.*status=(.*)'
+
+    @staticmethod
+    def calculate_columns_count_in_table(target_node, keyspace_name: str = 'keyspace1',
+                                         table_name: str = 'standard1') -> int:
+        query_cmd = f"SELECT * FROM {keyspace_name}.{table_name} LIMIT 1"
+        result = target_node.run_cqlsh(query_cmd)
+        return len(re.findall(r"(\| C\d+)", result.stdout))
+
+    @staticmethod
+    def get_random_item(items: list, pop: bool = False) -> Any:
+        if pop:
+            return items.pop(random.randint(0, len(items) - 1))
+
+        return items[random.randint(0, len(items) - 1)]
+
+    @classmethod
+    def distribute_test_files_to_cluster_nodes(cls, nodes, test_data: list) -> List:
+        """
+        Distribute test sstables over cluster nodes for `load-and-stream` test:
+        the feature allow loading arbitrary sstables that do not belong to a node into the cluster.
+        So we want to distribute sstables randomly.
+        Also add case when the 50% sstables of node2 (from test data cluster) will be loaded on the one node
+        and other 50% - on another
+        """
+        # Load 50% of node2's sstables to the different cluster nodes
+        map_files_to_node = [[test_data.pop(1), nodes[0]], [test_data.pop(1), nodes[1]]]
```

Comment:
test_data can contain from 1 record to 6 records

Reply:
done